Google.Api.Gax.Grpc/ApiServerStreamingCall.cs

```diff
@@ -107,5 +108,33 @@ public sealed class ApiServerStreamingCall<TRequest, TResponse>
             GaxPreconditions.CheckNotNull(valueSelector, nameof(valueSelector));
             return WithCallSettingsOverlay(request => CallSettings.FromGoogleRequestParamsHeader(parameterName, valueSelector(request)));
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="parameterName"></param>
+        /// <param name="extractionRegex"></param>
+        /// <param name="valueSelector"></param>
+        /// <returns></returns>
+        public ApiServerStreamingCall<TRequest, TResponse> WithExtractedGoogleRequestParam(string parameterName, Regex extractionRegex, Func<TRequest, string> valueSelector) =>
```

Comment:
How would you feel about creating the HeaderParameterExtractor in the generated code instead, so we only need to expose a single method here?